shipctl/x86_64/Ubuntu_16.04/utility.sh

```diff
@@ -689,6 +689,70 @@ put_resource_state_multi() {
   done
 }
 
+split_tests() {
+  if [ "$1" == "" ] || [ "$2" == "" ] || [ "$3" == "" ]; then
+    echo "Usage: shipctl split_tests TEST_PATH TEST_FILES_NAME_REGEX TEST_REPORTS_PATH"
+    exit 99
+  fi
+
+  # delete tmp files
+  rm -rf /tmp/current_tests.txt
+  rm -rf /tmp/past_test_timings.txt
+  rm -rf /tmp/sorted_past_test_timings.txt
+  rm -rf /tmp/sorted_past_tests.txt
+
+  # create tmp files
+  touch /tmp/current_tests.txt
+  touch /tmp/past_test_timings.txt
+  touch /tmp/sorted_past_test_timings.txt
+  touch /tmp/sorted_past_tests.txt
+
+  for i in "$(find $1 -name $2)";
+  do
+    echo -e "$i\n" >> /tmp/current_tests.txt;
+  done
+
+  find $3 -name \*.xml|while read i;
+  do
+    echo $(xq .testsuites.testsuite $i | jq -cr '.["@filepath"]," ", .["@time"]') >> /tmp/past_test_timings.txt
+  done
+
+  IFS=$'\r\n' GLOBIGNORE='*' command eval  'past_tests=($(cat /tmp/sorted_past_tests.txt))'
+  IFS=$'\r\n' GLOBIGNORE='*' command eval  'current_tests=($(cat /tmp/current_tests.txt))'
+  all_tests=()
+  current_job_tests=()
+  for i in "${past_tests[@]}";
+  do
+    skip=
+    for j in "${current_tests[@]}";
+    do
+      [[ $i == *$j ]] && { skip=1; break; }
```

Comment:
I'm not clear on what this does. Can you explain what's going on here?

Reply:
$i is the filepath that we are getting from the test results xmls:
```/root/src/github.com/ankul-shippable/railstest/test/01_test.rb```

$j is the filepath that we are getting from `find $test_path -name $test_files_name_regex`:
```test/01_test.rb```

I am adding `*` so that these two are matched.

Added sample contents for all the files used.

Reply:
may be we should rename `i` to `past_test` and `j` to `current_test`